sklearn/linear_model/ridge.py

```diff
@@ -281,6 +286,12 @@ def ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
     -----
     This function won't compute the intercept.
     """
+    if return_intercept and solver != 'sag':
+        warnings.warn("In Ridge, only 'sag' solver can currently fit the "
+                      "intercept when X is sparse. Solver has been "
+                      "automatically changed into 'sag'.")
+        solver = 'sag'
```

Comment:
You are right, I did not think about the public use of `ridge_regression`.
I'll change that